tests/setup_test.go

```diff
@@ -7008,3 +7008,143 @@ func TestProjectCommitStatusRedelivery(t *testing.T) {
 		}
 	})
 }
+
+func TestRunRequiredEnvVariables(t *testing.T) {
+	t.Parallel()
+
+	config := `
+		{
+			runs: [
+			{
+				name: 'run01',
+				tasks: [
+				{
+					name: 'task01',
+					runtime: {
+					containers: [
+						{
+						image: 'alpine/git',
+						},
+					],
+					},
+					steps: [
+					  { type: 'run', command: 'env -u AGOLA_SSHPRIVKEY' },
+					  { type: 'run', command: 'echo %d' },
+					],
+				},
+				],
+			},
+			],
+		}
+	`
+
+	tests := []struct {
+		name      string
+		config    string
+		env       map[string]string
+		runNumber uint64
+	}{
+		{
+			name:      "test push with run count 1",
+			config:    fmt.Sprintf(config, 1),
+			runNumber: 1,
+			env: map[string]string{
+				"AGOLA_RUN_COUNTER":   "",
+				"AGOLA_GIT_REF_TYPE":  "branch",
+				"AGOLA_GIT_REF":       "refs/heads/master",
+				"AGOLA_GIT_BRANCH":    "master",
+				"AGOLA_GIT_TAG":       "",
+				"AGOLA_GIT_COMMITSHA": "",
+			},
+		},
+	}
+
+	dir := t.TempDir()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sc := setup(ctx, t, dir, withGitea(true))
+	defer sc.stop()
+
+	giteaToken, tokenUser01 := createLinkedAccount(ctx, t, sc.gitea, sc.config)
+	gwClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, tokenUser01)
+
+	giteaAPIURL := fmt.Sprintf("http://%s:%s", sc.gitea.HTTPListenAddress, sc.gitea.HTTPPort)
+
+	giteaClient, err := gitea.NewClient(giteaAPIURL, gitea.SetToken(giteaToken))
+	testutil.NilError(t, err)
+
+	giteaRepo, project := createProject(ctx, t, giteaClient, gwClient, withVisibility(gwapitypes.VisibilityPrivate))
+
+	runCount := len(tests)
+
+	for _, tt := range tests {
+		push(t, tt.config, giteaRepo.CloneURL, giteaToken, "commit", false)
+	}
```

Comment:
It doesn't makes sense to push for every test. All of this should be put inside the tests loop (that could be parallelized).